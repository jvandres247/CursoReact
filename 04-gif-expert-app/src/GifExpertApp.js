import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ({defaultCategories = []}) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <div>
      <h2>Gif App Search</h2>
      <AddCategory setCategories={setCategories}/>
      <hr />
      <ol>
        {categories.map((category) => (
            <GifGrid key={category} category={category}/>
        ))}
      </ol>
    </div>
  );
};
