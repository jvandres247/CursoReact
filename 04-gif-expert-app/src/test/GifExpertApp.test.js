import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe("Pruebas en <GifExpertApp/>", () => {
  test("Debe de mostrarse correctamente", () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar una lista de categorias', () => {
      const categories = ['Naruto', 'Goku'];
      const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('GifGrid').length).toBe(categories.length);
  })
});
