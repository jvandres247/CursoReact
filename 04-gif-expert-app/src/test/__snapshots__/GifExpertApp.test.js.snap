// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> Debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    Gif App Search
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</div>
`;

exports[`Pruebas en <GifExpertApp/> Debe de mostrarse correctamente 1`] = `
<div>
  <h2>
    Gif App Search
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
